pennylane/pauli/dla/lie_closure.py

```diff
@@ -117,7 +122,23 @@ def lie_closure(
         >>> type(dla[0])
         pennylane.pauli.pauli_arithmetic.PauliSentence
 
+        In the case of sums of Pauli operators with many terms, it is often faster to use the matrix representation of the operators rather than
+        the semi-analytic :class:`~pennylane.pauli.PauliSentence` or :class:`~Operator` representation.
+        We can force this by using the ``matrix`` keyword. The resulting ``dla`` is a ``np.ndarray`` of dimension ``(dim_g, 2**n, 2**n)``, where ``dim_g`` is the
+        dimension of the DLA and ``n`` the number of qubits.
+
+        >>> dla = qml.lie_closure(ops, matrix=True)
+        >>> dla.shape
+        (6, 4, 4)
+
+        You can retrieve a semi-analytic representation again by using :func:`~pauli_decompose`.
+
+        >>> dla_ops = [qml.pauli_decompose(op) for op in dla]
```

Comment:
I get slightly different coefficients when using `qml.pauli_decompose`, does the minus sign matter?

```
>>> ops = [qml.X(0) @ qml.X(1), qml.Z(0), qml.Z(1)]
>>> qml.lie_closure(ops)
[X(0) @ X(1),
 Z(0),
 Z(1),
 -1.0 * (Y(0) @ X(1)),
 -1.0 * (X(0) @ Y(1)),
 Y(0) @ Y(1)]
```
```
>>> [qml.pauli_decompose(op) for op in qml.lie_closure(ops, matrix = True)]
[1.0 * (X(0) @ X(1)),
 1.0 * (Z(0) @ I(1)),
 1.0 * (I(0) @ Z(1)),
 1.0 * (Y(0) @ X(1)),
 1.0 * (X(0) @ Y(1)),
 1.0 * (Y(0) @ Y(1))]
```

Reply:
> does the minus sign matter?

Nope. But it may cause confusion so I added a small comment below this line to clarify that :)